src/purchases.ts

```diff
@@ -185,24 +195,25 @@ export default class Purchases {
    * Sets up Purchases with your API key and an app user id.
    * @param {String} apiKey RevenueCat API Key. Needs to be a String
    * @param {String?} appUserID An optional unique id for identifying the user. Needs to be a string.
-   * @param {boolean} [observerMode=false] An optional boolean. Set this to TRUE if you have your own IAP implementation and want to use only RevenueCat's backend. Default is FALSE.
-   * @param {boolean} [usesStoreKit2IfAvailable=false] DEPRECATED. An optional boolean. iOS-only. Defaults to FALSE. Setting this to TRUE will enable StoreKit2 on compatible devices.
-   * We recommend not using this parameter, letting RevenueCat decide for you which StoreKit implementation to use.
+   * @param {PURCHASES_ARE_COMPLETED_BY} [purchasesAreCompletedBy=PURCHASES_ARE_COMPLETED_BY.REVENUECAT] Set this to PURCHASES_ARE_COMPLETED_BY.MY_APP if you have your own IAP implementation and want to use only RevenueCat's backend. Default is PURCHASES_ARE_COMPLETED_BY.REVENUECAT.
+   * @param {STOREKIT_VERSION} [storeKitVersion=DEFAULT] iOS-only. Defaults to STOREKIT_2. StoreKit 2 is only available on iOS 16+. StoreKit 1 will be used for previous iOS versions regardless of this setting.
    * @param {ENTITLEMENT_VERIFICATION_MODE} [entitlementVerificationMode=ENTITLEMENT_VERIFICATION_MODE.DISABLED] Sets the entitlement verifciation mode to use. For more details, check https://rev.cat/trusted-entitlements
    * @param {boolean} [useAmazon=false] An optional boolean. Android-only. Set this to TRUE to enable Amazon on compatible devices.
    * @param {String?} userDefaultsSuiteName An optional string. iOS-only, will be ignored for Android.
    * Set this if you would like the RevenueCat SDK to store its preferences in a different NSUserDefaults suite, otherwise it will use standardUserDefaults.
    * Default is null, which will make the SDK use standardUserDefaults.
+   * @param {boolean} [pendingTransactionsForPrepaidPlansEnabled=false] An optional boolean. Android-only. Set this to TRUE to enable pending transactions for prepaid subscriptions in Google Play.
    */
   public static configure({
     apiKey,
     appUserID = null,
-    observerMode = false,
```

Comment:
I would also try to keep compatibility with observerMode but just deprecate it. I think the logic should follow:
- If `purchasesAreCompletedBy` provided, use that
- If not, and `observerMode` is provided with a non-null value, use the corresponding value for `purchasesAreCompletedBy`.
- Else null

Lmk what you think!

Reply:
We've completed removed all mention of `observerMode` in the native SDKs' configure functions, so I think it's appropriate to do it here too. It's a breaking change, but that should be okay since this is a major release!

Reply:
It's definitely tricky – in the iOS natives v5.0+, we now require you to provide storeKitVersion if you provide a value for `PurchasesAreCompletedBy` because the implementations between SK1 & SK2 are different. For that reason, we should actually consider making `storeKitVersion` a required parameter here as well 😅.

If we went with the approach you described earlier where both `observerMode` and `purchasesAreCompletedBy` are provided in the `configure`function, what would we do when the developer provides conflicting values (e.g., observerMode=true and purchasesAreCompletedBy=REVENUECAT)? Perhaps we should throw an error at runtime?

I agree that replacing the observerMode parameter with purchasesAreCompletedBy introduces a breaking change for developers, but it's already a breaking change in the iOS natives, so I'm not overly concerned with making this a breaking change for a major release.

Reply:
I agree the semantics and the implementation can become tricky if we keep both fields.

I can see two scenarios:
- We remove the field completely since the change to use `purchasesAreCompletedBy` is straightforward. One downside of this is the use would not get immediate feedback on what to change, they will just see a compilation error and will need to go check what changed and if there is an alternative field they can use.
- We deprecate and keep the `observerMode` field. We map it to the appropriate `purchasesAreCompletedBy`. If `purchasesAreCompletedBy` is present, prefer that. If both are present, log an error message.

I'd say the least disruptive for developers is 2, and since we already kept the field in `PurchasesConfiguration`, we can wait to remove it in the next major.

Note that even if `PurchasesConfiguration` still has the deprecated `observerMode` field, it is only used as the return value of the `.configure()` method, and we can still set the correct value there, by mapping from the `PurchasesAreCompletedBy` value.

Reply:
> what would we do when the developer provides conflicting values

I believe in this case `purchasesAreCompletedBy` would always have preference.

> we should actually consider making storeKitVersion a required parameter here as well

Hmm so we want to make it a required parameter when using `PurchasesAreCompletedAt.MY_APP` or `observerMode=true` right? I believe that would require some biggish changes in the API for configuration, since right now each parameter is independent... Maybe we can just log a warning message in PHC  in iOS if using `PurchasesAreCompletedAt.MY_APP` or `observerMode=true` and not specifying `storeKitVersion`?